.devops/deploy-pipelines.yml

```
@@ -413,105 +576,104 @@ stages:
                     script: |
                       helm repo add microservice-chart https://pagopa.github.io/aks-microservice-chart-blueprint
                       helm dep build helm
-                - template: azure-templates/helm-microservice-chart-deploy.yml
-                  parameters:
-                    DO_DEPLOY: true
-                    DO_BLUE_GREEN_DEPLOY: true
-                    ENV: 'PROD'
-                    KUBERNETES_SERVICE_CONN: $(PROD_KUBERNETES_SERVICE_CONN)
-                    NAMESPACE: ecommerce
-                    APP_NAME: $(K8S_IMAGE_REPOSITORY_NAME)
-                    VALUE_FILE: "helm/values-prod.yaml"
-                    GREEN_VERSION: $(app_version)
-                    BLUE_VERSION: $(Build.SourceVersion)
+                - task: HelmDeploy@0
+                  condition: succeeded()
+                  displayName: Deploy current PROD version $(currentVersion) as STAGING release
+                  inputs:
+                    kubernetesServiceEndpoint: $(PROD_KUBERNETES_SERVICE_CONN)
+                    namespace: ecommerce
+                    command: upgrade
+                    chartType: filepath
+                    chartPath: helm
+                    chartName: $(K8S_IMAGE_REPOSITORY_NAME)-stage
+                    releaseName: $(K8S_IMAGE_REPOSITORY_NAME)-stage
+                    valueFile: helm/values-prod.yaml
+                    install: true
+                    waitForExecution: true
+                    arguments: --timeout 5m0s
+                    overrideValues: microservice-chart.ingress.path=/pagopa-ecommerce-transactions-scheduler-service-staging/(.*),microservice-chart.envConfig.DEPLOYMENT_VERSION=STAGING,microservice-chart.fullnameOverride=pagopaecommercetxschedulerservice-staging
```

```suggestion
                    overrideValues: microservice-chart.ingress.path=/pagopa-ecommerce-transactions-scheduler-service-staging/(.*),microservice-chart.envConfig.DEPLOYMENT_VERSION=STAGING,microservice-chart.fullnameOverride=pagopaecommercetxschedulerservice-staging,microservice-chart.envConfig.PENDING_TRANSACTIONS_SCHEDULE_CRON=-
```

here we are deploying the scheduler staging version.
Since scheduled batches are not yet implementing a semaphore for exclusive execution this configuration can lead to a batch being executed twice (by staging and prod deployments) while performing rolling update (if rolling update is performed near to an execution time).

Adding this override we are [disabling batch](http://localhost:63342/pagopa-ecommerce-transactions-scheduler-service/spring-context-5.3.19-javadoc.jar/org/springframework/scheduling/annotation/Scheduled.html?_ijt=ag2p7fjlr3begonvc50rq3a756&_ij_reload=RELOAD_ON_SAVE#CRON_DISABLED) for staging release avoiding that batch can be spawn multiple times during rolling update